packages/perseus/src/types.ts

```diff
@@ -262,7 +262,7 @@ export type APIOptions = Readonly<{
      * This is useful for when we need to know how a widget has shuffled its
      * the available choices.
      */
-    onWidgetStartProps?: (widgets: PerseusWidgetsMap) => PerseusWidgetsMap;
+    onWidgetStartProps?: (widgets: PerseusWidgetsMap) => void;
```

Comment:
It's a bit confusing because the callback doesn't use the returned data in any way. `void` makes that clear, `unknown` leaves it ambiguous.

Reply:
I'm not sure why we would want to allow callers to return data in the callback if it serves no purpose.

It looks like you are able to manipulate the data before sending it through to Perseus, which is not the case. For the sake of it being self-documenting, `void` is the correct type here.

Reply:
If you are aware of some future case where this callback will allow callers to intercept and manipulate widget props, I'll just close the PR rather than make it `unknown`, which is not an improvement in either case (self-documenting about current behavior or leaving the door open for mutating the props in the future)

Reply:
Dug up some documentation. `unknown` is also clearly defined here.

"void represents the return value of functions which don’t return a value."

https://www.typescriptlang.org/docs/handbook/2/functions.html#void